modules/gdscript/editor/script_templates/CompositorEffect/post_process_shader.gd

```diff
@@ -0,0 +1,149 @@
+@tool
+extends CompositorEffect
+class_name PostProcessShader
+
+const template_shader: String = "#version 450
+
+// Invocations in the (x, y, z) dimension
+layout(local_size_x = 8, local_size_y = 8, local_size_z = 1) in;
+
+layout(rgba16f, set = 0, binding = 0) uniform image2D color_image;
+
+// Our push constant
+layout(push_constant, std430) uniform Params {
+	vec2 raster_size;
+	vec2 reserved;
+} params;
+
+// The code we want to execute in each invocation
+void main() {
+	ivec2 uv = ivec2(gl_GlobalInvocationID.xy);
+	ivec2 size = ivec2(params.raster_size);
+
+	if (uv.x >= size.x || uv.y >= size.y) {
+		return;
+	}
+
+	vec4 color = imageLoad(color_image, uv);
+
+	#COMPUTE_CODE
+
+	imageStore(color_image, uv, color);
+}"
+
+@export_multiline var shader_code: String = "":
+	set(value):
+		mutex.lock()
+		shader_code = value
+		shader_is_dirty = true
+		mutex.unlock()
+
+var rd: RenderingDevice
+var shader: RID
+var pipeline: RID
+
+var mutex: Mutex = Mutex.new()
+var shader_is_dirty: bool = true
+
+# Called when this resource is constructed.
+func _init():
+	effect_callback_type = EFFECT_CALLBACK_TYPE_POST_TRANSPARENT
+	rd = RenderingServer.get_rendering_device()
+
+# System notifications, we want to react on the notification that
+# alerts us we are about to be destroyed.
+func _notification(what):
+	if what == NOTIFICATION_PREDELETE:
+		if shader.is_valid():
+			# Freeing our shader will also free any dependents such as the pipeline!
+			RenderingServer.free_rid(shader)
+
+# Check if our shader has changed and needs to be recompiled.
+func _check_shader() -> bool:
+	if not rd:
+		return false
+
+	var new_shader_code: String = ""
+
+	# Check if our shader is dirty.
+	mutex.lock()
+	if shader_is_dirty:
+		new_shader_code = shader_code
+		shader_is_dirty = false
+	mutex.unlock()
+
+	# We don't have a (new) shader?
+	if new_shader_code.is_empty():
+		return pipeline.is_valid()
+
+	# Apply template.
+	new_shader_code = template_shader.replace("#COMPUTE_CODE", new_shader_code);
+
+	# Out with the old.
+	if shader.is_valid():
+		rd.free_rid(shader)
+		shader = RID()
+		pipeline = RID()
+
+	# In with the new.
+	var shader_source: RDShaderSource = RDShaderSource.new()
+	shader_source.language = RenderingDevice.SHADER_LANGUAGE_GLSL
+	shader_source.source_compute = new_shader_code
+	var shader_spirv: RDShaderSPIRV = rd.shader_compile_spirv_from_source(shader_source)
+
+	if shader_spirv.compile_error_compute != "":
+		push_error(shader_spirv.compile_error_compute)
+		push_error("In: " + new_shader_code)
```

Comment:
```suggestion
		push_error(shader_spirv.compile_error_compute + "\nIn: " + new_shader_code)
```

`push_error()` doesn't just print an error like `printerr()`, it creates an expandable entry with a stack trace. So it should be one call per error.